sdk/go/common/testing/environment.go

```diff
@@ -152,8 +153,15 @@ func (e *Environment) PathExists(p string) bool {
 	return err == nil
 }
 
+var YarnInstallMutex sync.Mutex
```

Comment:
I was initially slightly hesitant about publicly exporting this, but I think it's fine.

We've used `yarn` pretty pervasively from early on, but `npm` has evolved quite a bit since then. I've always wondered if we should consider moving away from `yarn` and use standard `npm` instead. (I wonder if `npm` even has this timeout issue). I guess if we do ever get around to switching to `npm`, we could still keep this publicly exported variable, we just wouldn't be using it anymore, since we would be running `npm` rather than `yarn` (unless there's a need for a similar mutex for `npm` and we co-opt this mutex for that purpose as well).

To that end, should we use a more general name, like `NodeInstallMutex`, or is it better to keep it yarn-specific since right now it really is only for any `yarn` command? (I don't feel strongly either way and am OK with `YarnInstallMutex`; I just wanted to raise the question in case anyone else has a stronger opinion).

Reply:
I think it should be Yarn specific, we're working around a specific behavior with the mutex & timeout that `yarn` implements. If npm, pnpm, or other packagers require a similar lock, we should separate them out into individual mutexes so that we can still parallelize.